### ISSUE TITLE
Should we raise `UNSUPPORTED_API` if priviledged addon?

### ISSUE THREAD

**Issue by leplatrem** [mozilla/addons-linter#4351]
### Describe the problem and steps to reproduce it:

Our Remote Settings DevTools extension defines an Experiments API in the `manifest.json`:
https://github.com/mozilla-extensions/remote-settings-devtools/blob/3e6463c56ff5b2db401794dcee47221c0a8193cb/extension/manifest.json#L34-L43

But the linter still warns about an `UNSUPPORTED_API` in `content/script.js`.

```
[task 2022-05-24T16:52:12.500Z] UNSUPPORTED_API     experiment…   This API has not been   content/sc…   7      24    
[task 2022-05-24T16:52:12.500Z]                     s.remotese…   implemented by          ript.js                    
[task 2022-05-24T16:52:12.500Z]                     ttings is     Firefox.                                           
[task 2022-05-24T16:52:12.500Z]                     not                                                              
[task 2022-05-24T16:52:12.501Z]                     supported           
```

### What happened?

The ability to define new experiments API is reserved to priviledged addons. And the linter does not seem aware of this. (similar to #4334 ?)

### What did you expect to happen?

The linter should not warn about an unsupported API if a priviledged addon is built and the experiment API is defined in the manifest.

**Comment by willdurand** [comment]
Thanks for filing this issue Mathieu. That's also something I was wondering and I think we could silent this warning because it does not make much sense when `--privileged` is passed anyway. We don't have extra validation.

**Comment by ioanarusiczki** [comment]
@willdurand I ran locally linter 5.8.0 for the remote-settings-devtools addon and when I add --privileged flag, the yellow warning about the API is no longer displayed.

Validation summary: 

![validation results](https://user-images.githubusercontent.com/33448286/173546966-9f61ea13-33c5-40ed-8b35-fc6b7ea17da3.png)
